File: 27.移除元素.c
/*
 * @lc app=leetcode.cn id=27 lang=c
 *
 * [27] 移除元素
 */

// @lc code=start

int removeElement(int *nums, int numsSize, int val)
{
    if (numsSize == 0) return 0;
    if (numsSize == 1)
    {
        if (nums[0] == val)
        {
            return 0;
        }
        else
        {
            return 1;
        }

    }

    int temp = nums[0], p = 0;

    for (int i = 1; i < numsSize; i++)
    {
        if (nums[i] != val)
        {
            nums[p++] = nums[i];
        }
    }
    if (temp != val)
    {
        nums[p++] = temp;
    }
    return p;
}
// @lc code=end

File: 27.移除元素.rs
/*
 * @lc app=leetcode.cn id=27 lang=rust
 *
 * [27] 移除元素
 */

// @lc code=start
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut point:usize = 0;
        
        for index in 0..nums.len() {
            if nums[index] != val {
                nums[point] = nums[index];
                point += 1;
            }
        }
        point as i32  //少了return  内存使用减少了0.3MB
    }
}
// @lc code=end

